aws-lc-rs/src/hkdf.rs

```diff
@@ -128,24 +124,16 @@ impl Salt {
     // * `value.len() > 0` is true
     //
     /// # Panics
-    /// `new` panics if the salt length exceeds the limit
+    /// `new` panics if salt creation fails
     #[must_use]
     pub fn new(algorithm: Algorithm, value: &[u8]) -> Self {
-        Salt::try_new(algorithm, value).expect("Salt length limit exceeded.")
+        Salt::try_new(algorithm, value).expect("Failed to create Salt")
     }
 
+    #[allow(clippy::unnecessary_wraps)]
     fn try_new(algorithm: Algorithm, value: &[u8]) -> Result<Salt, Unspecified> {
-        let salt_len = value.len();
-        if salt_len > MAX_HKDF_SALT_LEN {
-            return Err(Unspecified);
-        }
-        let mut salt_bytes = [0u8; MAX_HKDF_SALT_LEN];
-        salt_bytes[0..salt_len].copy_from_slice(value);
-        Ok(Self {
-            algorithm,
-            bytes: salt_bytes,
-            len: salt_len,
-        })
+        let bytes = value.to_vec().into_boxed_slice();
+        Ok(Self { algorithm, bytes })
```

Comment:
I don't think `try_new` is needed now that we have no effective limit, we can just move this implementation into `new`.